python-avd/pyavd/_schema/models/path.py

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2024 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Iterator
+
+
+class Path(list):
+    class Key(str):
+        __slots__ = ()
```

Comment:
These classes were experiments. I will remove them. Type alias would not be good here (since all aliases of str would be treated the same), but using `newtype()` would be more optimal.